datafusion/expr/src/built_in_function.rs

```diff
@@ -963,15 +967,20 @@ impl BuiltinScalarFunction {
             BuiltinScalarFunction::ArrayPosition => {
                 Signature::variadic_any(self.volatility())
             }
-            BuiltinScalarFunction::ArrayPositions => Signature::any(2, self.volatility()),
-            BuiltinScalarFunction::ArrayPrepend => Signature {
-                type_signature: ElementAndArray,
-                volatility: self.volatility(),
-            },
+            BuiltinScalarFunction::ArrayPositions => {
+                Signature::array_and_element(self.volatility())
+            }
+            BuiltinScalarFunction::ArrayPrepend => {
+                Signature::element_and_array(self.volatility())
+            }
             BuiltinScalarFunction::ArrayRepeat => Signature::any(2, self.volatility()),
```

Comment:
Although the functions including `array_repeat`, `array_union`, `array_intersect`, and `array_except` are also two-argument, `Signature::array_and_element` could not handle `NULL` well for them. The following PR could continue to work on this issue.